File: app/src/main/java/activitytest/example/com/roomdemo/home/adapter/BlogAdapter.java
package activitytest.example.com.roomdemo.home.adapter;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

import activitytest.example.com.roomdemo.R;
import activitytest.example.com.roomdemo.databinding.BlogItemCardBinding;
import activitytest.example.com.roomdemo.main.activity.WebViewActivity;
import activitytest.example.com.roomdemo.home.vo.BlogIntroduce;
import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.recyclerview.widget.RecyclerView;


public class BlogAdapter extends RecyclerView.Adapter<BlogAdapter.BlogInnerView> {

    private final Context context;
    private final List<BlogIntroduce> blogIntroduceList;

    public BlogAdapter(Context context,List<BlogIntroduce> blogIntroduceList) {
        this.context = context;
        this.blogIntroduceList = blogIntroduceList;
    }

    @NonNull
    @Override
    public BlogInnerView onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        BlogItemCardBinding viewDataBinding = DataBindingUtil.inflate ( LayoutInflater.from ( parent.getContext () ), R.layout.blog_item_card, parent, false );
        return new BlogInnerView ( viewDataBinding );

    }

    @Override
    public void onBindViewHolder(@NonNull BlogInnerView holder, int position) {

        holder.blogItemCardBinding.blogTags.setText ( blogIntroduceList.get ( position ).getTag () );
        holder.blogItemCardBinding.blogTitle.setText ( blogIntroduceList.get ( position ).getBlogTitle () );
        holder.blogItemCardBinding.blogContent.setText ( blogIntroduceList.get ( position ).getTime () );

        holder.itemView.setOnClickListener ( new View.OnClickListener () {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent (context, WebViewActivity.class );

                intent.putExtra ( "html",blogIntroduceList.get ( position ).getHtml () );

                context.startActivity ( intent );
            }
        } );
    }



    @Override
    public int getItemCount() {
        return blogIntroduceList.size ();
    }

    public static class BlogInnerView extends RecyclerView.ViewHolder{

        BlogItemCardBinding blogItemCardBinding;

        public BlogInnerView(@NonNull BlogItemCardBinding itemView) {
            super ( itemView.getRoot () );
            blogItemCardBinding = itemView;

        }
    }


}

File: app/src/main/java/activitytest/example/com/roomdemo/home/adapter/BlogAdapter.kt
package activitytest.example.com.roomdemo.home.adapter

import activitytest.example.com.roomdemo.R
import activitytest.example.com.roomdemo.databinding.HomeCardItemBinding
import activitytest.example.com.roomdemo.home.adapter.BlogAdapter.BlogInnerView
import activitytest.example.com.roomdemo.home.utils.MonthUtil
import activitytest.example.com.roomdemo.home.utils.TagsUtil
import activitytest.example.com.roomdemo.home.vo.BlogIntroduce
import activitytest.example.com.roomdemo.main.activity.WebViewActivity
import activitytest.example.com.roomdemo.main.database.entity.Tags
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView

class BlogAdapter(private val context: Context, private val blogIntroduceList: List<BlogIntroduce>) : RecyclerView.Adapter<BlogInnerView>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BlogInnerView {
        val viewDataBinding: HomeCardItemBinding = DataBindingUtil.inflate(LayoutInflater.from(parent.context), R.layout.home_card_item, parent, false)
        return BlogInnerView(viewDataBinding)
    }

    override fun onBindViewHolder(holder: BlogInnerView, position: Int) {
        val tag = TagsUtil.getTag(blogIntroduceList[position].tags)
        holder.blogItemCardBinding.blogTags.text = tag
        holder.blogItemCardBinding.blogTitle.text = blogIntroduceList[position].blogTitle
        holder.blogItemCardBinding.blogContent.text =  blogIntroduceList[position].time.let { MonthUtil.getNowDate(it) }
        holder.itemView.setOnClickListener {
            val intent = Intent(context, WebViewActivity::class.java)
            intent.putExtra("html", blogIntroduceList[position].html)
            context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int {
        return blogIntroduceList.size
    }

    class BlogInnerView(var blogItemCardBinding: HomeCardItemBinding) : RecyclerView.ViewHolder(blogItemCardBinding.root)
}